framework/include/ics/FunctionIC.h

```diff
@@ -35,6 +35,12 @@ class FunctionIC : public InitialCondition
 
   FunctionIC(const InputParameters & parameters);
 
+  /**
+   * Get the function name
+   * @return function name
+   */
+  const FunctionName & functionName() const { return _function_name; }
```

Comment:
No need to store `_function_name`, remove that member and just do

```
const FunctionName &
FunctionIC::functionName() const
{
  return _func.name();
}
```

in source